File: typescript/examples/02_verify_receipt.ts
/**
 * Parmana TypeScript SDK
 *
 * Example 02
 *
 * Verify Receipt
 */

import {
  ExecutionTrustRecord,
  ParmanaClient,
} from "../src/index.js";

async function main(): Promise<void> {
  const client = new ParmanaClient({
    endpoint: "http://localhost:8080",
  });

  /**
   * Placeholder Trust Record.
   *
   * In production this would be returned by the
   * Parmana Runtime after execution.
   */
  const trustRecord = {
    trustRecordId: "trust-record-001",
    businessTransactionId: "txn-001",

    transaction: {} as never,

    overrides: [],

    executions: [],

    verifications: [],

    receipts: [],

    trustRecordHash:
      "7f84e2d3c1d9d8d3d2c8f7e9b6a4c5d2",

    createdAt: new Date(),

    updatedAt: new Date(),
  } satisfies ExecutionTrustRecord;

  console.log();
  console.log("======================================");
  console.log(" Parmana TypeScript SDK");
  console.log(" Example 02 - Verify Receipt");
  console.log("======================================");
  console.log();

  console.log("Trust Record");
  console.log("-------------------------");
  console.log(
    "Trust Record ID :",
    trustRecord.trustRecordId,
  );
  console.log(
    "Transaction ID  :",
    trustRecord.businessTransactionId,
  );
  console.log(
    "Hash            :",
    trustRecord.trustRecordHash,
  );
  console.log();

  try {
    const verification =
      await client.verify(trustRecord);

    console.log("Verification");
    console.log("-------------------------");
    console.log(
      "Verification ID :",
      verification.verificationId,
    );
    console.log(
      "Status          :",
      verification.status,
    );
    console.log(
      "Verified At     :",
      verification.verifiedAt,
    );
  } catch (error) {
    console.log();
    console.error(
      "Verification service not available (expected during SDK development).",
    );

    if (error instanceof Error) {
      console.error(error.message);
    }
  }

  console.log();
  console.log("Example completed.");
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: python-old/examples/02_verify_receipt.py
"""
Parmana Python SDK

Example 02

Verify Receipt

Demonstrates verification of a Parmana
Execution Trust Receipt.
"""

from datetime import datetime
from uuid import uuid4

from parmana import ParmanaClient

from parmana.models.receipt import Receipt


def main() -> None:
    """
    Verify a Parmana receipt.
    """

    receipt = Receipt(
        receipt_id=str(uuid4()),
        business_transaction_id=str(uuid4()),
        execution_id=str(uuid4()),
        trust_record_hash="9d6e0a0e0f4d2d6b8a1f9f3c5d7e8a9b",
        receipt_hash="b6d5d8f92dcd9dbeea21e1f9a5c91b66",
        signature="ed25519-signature",
        algorithm="Ed25519",
        issued_at=datetime.utcnow(),
    )

    client = ParmanaClient()

    #
    # Placeholder verification.
    #
    verification = client.verification.verify(receipt)

    print()

    print("===================================")
    print("Parmana Receipt Verification")
    print("===================================")

    print(f"Receipt ID          : {receipt.receipt_id}")
    print(f"Transaction ID      : {receipt.business_transaction_id}")
    print(f"Trust Record Hash   : {receipt.trust_record_hash}")

    print()

    print("Verification")

    print("----------------------------")

    print(f"Verification ID : {verification.verification_id}")
    print(f"Status          : {verification.status.value}")
    print(f"Message         : {verification.message}")
    print(f"Verified At     : {verification.verified_at}")

    print()

    print("Receipt verification completed.")

    print("===================================")


if __name__ == "__main__":
    main()